packages/core/src/common/object.spec.ts

```diff
@@ -0,0 +1,103 @@
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { expect } from 'chai';
+import { deepFreeze, notEmpty, isEmpty, deepClone } from './objects';
+
+describe.only('Object', () => {
+
+    describe('#deepClone', () => {
+
+        describe('primitives', () => {
+            it('should deep clone numbers', () => {
+                expect(1).to.equal(deepClone(1));
+            });
+            it('should deep clone strings', () => {
+                expect('foo').to.equal(deepClone('foo'));
+                expect('').to.equal(deepClone(''));
+            });
+            it('should deep clone booleans', () => {
+                expect(true).to.equal(deepClone(true));
+                expect(false).to.equal(deepClone(false));
+            });
+        });
+
+        describe('undefined', () => {
+            it('should deep clone undefined', () => {
+                expect(undefined).to.equal(deepClone(undefined));
+            });
+        });
+
+        describe('arrays', () => {
+            it('should deep clone arrays', () => {
+                const a = [1, 2, 3];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+
+            it('should deep clone nested arrays', () => {
+                const a = [[1], [2]];
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(Array.isArray(a)).to.equal(true);
+                expect(Array.isArray(b)).to.equal(true);
+            });
+        });
+
+        describe('objects', () => {
+            it('should deep clone objects', () => {
+                const a = { 'foo': true, 'bar': 1 };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+
+            it('should deep clone nested objects', () => {
+                const a = { 'foo': true, 'nested': { 'foo': 1 } };
+                const b = deepClone(a);
+                expect(a).to.deep.equal(b);
+                expect(typeof a).to.equal('object');
+                expect(typeof b).to.equal('object');
+            });
+        });
+    });
+
+    describe('#deepFreeze', () => {
+        it('should deep freeze an object', () => {
+            const e = { valeur: 10, boolval: true };
+            const d = deepFreeze(e);
+            expect(true).to.equal(Object.isFrozen(d));
+        });
```

Comment:
Given the object is frozen, we can write a test like the following where we expect it to complain when we update a property:

```ts
describe('#deepFreeze', () => {
    it('should deep freeze an object', () => {
        const value = 10;
        const e = { foo: value };
        const d = deepFreeze(e);
        expect(Object.isFrozen(d)).to.equal(true);
        expect(() => d.foo = value + 1).to.throw(TypeError);
        expect(d.foo).to.equal(value);
    });
});
```

Reply:
@EstFoisy I do not believe this comment was addressed?